airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/JobCreationAndStatusUpdateActivityImpl.java

```diff
@@ -288,10 +292,11 @@ public void jobCancelled(final JobCancelledInput input) {
       jobPersistence.cancelJob(jobId);
 
       final Job job = jobPersistence.getJob(jobId);
-      trackCompletion(job, JobStatus.FAILED);
       emitJobIdToReleaseStagesMetric(OssMetricsRegistry.JOB_CANCELLED_BY_RELEASE_STAGE, jobId);
       jobNotifier.failJob("Job was cancelled", job);
+      trackCompletion(job, JobStatus.FAILED);
     } catch (final IOException e) {
```

Comment:
Should we catch `Throwable` here as well?

Reply:
@gosusnp Probably.  Let's see if we even hit the catch block now that the new recording method has been added.  We can and most likely will need to make more changes to this. cc:  @colesnodgrass